src/main/java/com/mongodb/jdbc/logging/MongoLoggerUtils.java

```diff
@@ -0,0 +1,95 @@
+package com.mongodb.jdbc.logging;
+
+import com.mongodb.MongoException;
+import com.mongodb.jdbc.MongoConnection;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.*;
+
+public class MongoLoggerUtils {
+    private static ConcurrentHashMap<Integer, Logger> loggerPerConnection =
+            new ConcurrentHashMap<>();
+
+    public static Logger initConnectionLogger(Integer connection_id, Level logLevel, File logDir) {
+        Logger parentConnectionLogger = loggerPerConnection.get(connection_id);
+        if (null != parentConnectionLogger) {
+            // There is already a parent connection logger, there is a problem
+            throw new MongoException("Parent connection logger already initialed. Ignored.");
```

Comment:
I'm not sure if this should be a `MongoException`, but I can't think of anything better, either. I do slightly worry that this will look like a mongo java driver exception rather than a mongo jdbc exception. Maybe we should create a `MongoJDBCException` `RuntimeException`?